### PR TITLE
    [REF] pos*: remove temp screens

### PR DESCRIPTION
**Pull request by vlst-odoo**
The pos uses a special mechanism to handle the so called "temp screens".
    These are screens that are not considered part of the normal flow.

    This mechanism is overly complicated and error prone.

    Apart from this, the whole idea of temp screens is not really needed,
    as these screens can be considered as normal screens, or as dialogs.

    This is what we do in this commit: we remove the temp screens mechanism
    and adapt all the temps screens to be normal screens or dialogs.

    Task 3613049
https://github.com/odoo/enterprise/pull/51561

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/point_of_sale/static/src/app/store/db.js [hunk 1]
```diff
@@ -430,7 +430,12 @@ export class PosDB {
                 this.partner_by_barcode[partner.barcode] = partner;
             }
             updated[partner.id] = partner;
-            this.partner_by_id[partner.id] = partner;
+            // this.partner_by_id[partner.id] = partner;
```
Comment (vlst-odoo) [inline addons/point_of_sale/static/src/app/store/db.js:433]: remove comment

File: addons/pos_restaurant/static/src/overrides/models/pos_store.js [hunk 2]
```diff
@@ -50,7 +50,6 @@ patch(PosStore.prototype, {
     },
     async actionAfterIdle() {
         if (document.querySelector(".modal-open")) {
-            this.closeTempScreen();
```
Comment (vlst-odoo) [inline addons/pos_restaurant/static/src/overrides/models/pos_store.js:54]: add something like `showscreen('floorscreen')`

File: addons/point_of_sale/static/src/app/screens/partner_list/partner_editor/partner_editor.xml [hunk 3]
```diff
@@ -1,76 +1,80 @@
...
-                    <input class="detail partner-name form-control form-control-lg" t-model="changes.name" name="name" placeholder="Name" />
-                </div>
-                <div class="partner-details-box row row-cols-1 row-cols-sm-2 gy-3 mt-3">
-                    <t t-foreach="['Street', 'City', 'Zip', 'Email', 'Phone', 'Mobile', 'Barcode' ]" t-as="item" t-key="item">
-                        <div class="partner-detail col">
-                            <label class="form-label label" t-attf-for="{{item}}" t-esc="item"/>
-                            <input class="detail form-control" t-attf-id="{{item}}" t-attf-name="{{item}}" t-model="changes[item.toLowerCase()]" t-attf-placeholder="{{item}}" t-att-class="{'border-danger': missingFields.includes(item.toLowerCase())}" />
-                        </div>
+    <t t-name="point_of_sale.PartnerEditor">
+        <Dialog title="'Edit Partner'" bodyClass="'overflow-y-auto'" contentClass="'h-100'">
```
Comment (vlst-odoo) [inline addons/point_of_sale/static/src/app/screens/partner_list/partner_editor/partner_editor.xml:4]: translate this everywhere

File: addons/point_of_sale/static/src/app/screens/partner_list/partner_line/partner_line.js [hunk 4]
```diff
@@ -4,11 +4,5 @@ import { Component } from "@odoo/owl";
...
 export class PartnerLine extends Component {
     static template = "point_of_sale.PartnerLine";
-
-    get highlight() {
-        return this._isPartnerSelected ? "highlight active" : "";
-    }
-    get _isPartnerSelected() {
-        return this.props.partner === this.props.selectedPartner;
-    }
+    // static props = ["partner", "isSelected"];
```
Comment (vlst-odoo) [inline addons/point_of_sale/static/src/app/screens/partner_list/partner_line/partner_line.js:7]: fix this props

File: addons/point_of_sale/static/src/app/screens/product_screen/control_buttons/select_partner_button/select_partner_button.js [hunk 5]
```diff
@@ -6,7 +6,7 @@ import { useService } from "@web/core/utils/hooks";
 
 export class SelectPartnerButton extends Component {
     static template = "point_of_sale.SelectPartnerButton";
-    static props = ["partner"];
+    static props = ["partner?"];
```
Comment (vlst-odoo) [inline addons/point_of_sale/static/src/app/screens/product_screen/control_buttons/select_partner_button/select_partner_button.js:9]: move this to the ctrl button pr

File: addons/point_of_sale/static/tests/tours/helpers/ProductScreenTourMethods.js [hunk 6]
```diff
@@ -104,23 +105,13 @@ export function clickPartnerButton() {
         },
         {
             content: "partner screen is shown",
-            trigger: ".pos-content .partnerlist-screen",
+            trigger: PartnerList.clickPartner().trigger,
```
Comment (caburj) [inline addons/point_of_sale/static/tests/tours/helpers/ProductScreenTourMethods.js:108]: shouldn't there be `in_modal` key?

File: addons/pos_hr/static/src/app/header_lock_button/header_lock_button.js
Comment (caburj) [inline addons/pos_hr/static/src/app/header_lock_button/header_lock_button.js:1]: What am I missing, why is this removed?

File: addons/pos_restaurant/static/src/app/bill_screen/bill_screen.js [hunk 7]
```diff
@@ -1,29 +1,27 @@
...
     static template = "pos_restaurant.BillScreen";
-    static components = { OrderReceipt };
-    confirm() {
-        if (!this.env.isMobile) {
-            this.props.resolve({ confirmed: true, payload: null });
-            this.pos.closeTempScreen();
-        }
+    static components = { OrderReceipt, Dialog };
+    setup() {
+        this.state = useState({ printing: false });
```
Comment (caburj) [inline addons/pos_restaurant/static/src/app/bill_screen/bill_screen.js:13]: `this.state` doesn't seem to be useful.

File: addons/pos_restaurant/static/src/app/bill_screen/bill_screen.js [hunk 8]
```diff
@@ -1,29 +1,27 @@
...
-    get isBill() {
-        return true;
+        this.pos.get_order()._printed = await this.printer.print(
+            OrderReceipt,
+            {
+                data: this.pos.get_order().export_for_printing(),
+                formatCurrency: this.env.utils.formatCurrency,
+            },
+            { webPrintFallback: true }
+        );
```
Comment (caburj) [inline addons/pos_restaurant/static/src/app/bill_screen/bill_screen.js:25]: why are you doing this? This is not part of the logic. `_printed` should only be true if the order is validated.

Comment (vlst-odoo) [reply addons/pos_restaurant/static/src/app/bill_screen/bill_screen.js:25]: indeed

File: addons/pos_restaurant/static/src/app/bill_screen/bill_screen.js
Comment (caburj) [inline addons/pos_restaurant/static/src/app/bill_screen/bill_screen.js:1]: What happened to the auto print? If auto printing is activated, the billscreen shouldn't show, the receipt will just automatically be printed.

Comment (caburj) [reply addons/pos_restaurant/static/src/app/bill_screen/bill_screen.js:1]: Oh nevermind. It's still there.

File: addons/pos_restaurant/static/src/overrides/models/pos_store.js (hunk 2 above)
Comment (caburj) [reply addons/pos_restaurant/static/src/overrides/models/pos_store.js:53]: I think you need to do this.

File: addons/pos_hr/static/src/app/header_lock_button/header_lock_button.js
Comment (vlst-odoo) [reply addons/pos_hr/static/src/app/header_lock_button/header_lock_button.js:1]: it's moved to the navbar component

File: addons/pos_restaurant/static/tests/tours/ControlButtons.tour.js [hunk 9]
```diff
@@ -46,9 +45,9 @@ registry.category("web_tour.tours").add("ControlButtonsTour", {
 
             // Test PrintBillButton
             ProductScreen.controlButton("Bill"),
-            Dialog.confirm(),
-            BillScreen.isShown(),
-            BillScreen.clickOk(),
+            // Dialog.confirm(),
```
Comment (caburj) [inline addons/pos_restaurant/static/tests/tours/ControlButtons.tour.js:48]: remove this comment

File: addons/point_of_sale/static/src/app/screens/partner_list/partner_list.xml [hunk 10]
```diff
@@ -1,80 +1,50 @@
...
+                        <th class="py-2"></th>
+                    </tr>
+                </thead>
+                <tbody>
+                    <PartnerLine t-foreach="partners" t-as="partner" t-key="partner.id"
+                                partner="partner"
+                                isSelected="props.partner?.id === partner.id"
+                                isBalanceDisplayed="isBalanceDisplayed"
+                                onClickEdit.bind="editPartner"
+                                onClickUnselect.bind="() => this.clickPartner()"
```
[the remark this answers is not in the document]
Comment (vlst-odoo) [reply addons/point_of_sale/static/src/app/screens/partner_list/partner_list.xml:37]: indeed

File: addons/point_of_sale/static/src/app/store/pos_store.js [hunk 11]
```diff
@@ -1649,10 +1649,11 @@ export class PosStore extends Reactive {
         ]);
     }
     showScreen(name, props) {
+        this.previousScreen = this.mainScreen.component?.name;
         const component = registry.category("pos_screens").get(name);
         this.mainScreen = { component, props };
         // Save the screen to the order so that it is shown again when the order is selected.
-        if (component.storeOnOrder ?? true) {
+        if ((component.storeOnOrder ?? true) && name !== "LoginScreen") {
```
[the remark this answers is not in the document]
Comment (vlst-odoo) [reply addons/point_of_sale/static/src/app/store/pos_store.js:1686]: that's correct. I now rely on the `storeOnOrder` property. This is much better

File: addons/point_of_sale/static/src/app/screens/receipt_screen/receipt_screen.js [hunk 12]
```diff
@@ -120,23 +119,14 @@ export class ReceiptScreen extends Component {
...
             {
                 data: this.pos.get_order().export_for_printing(),
                 formatCurrency: this.env.utils.formatCurrency,
             },
             { webPrintFallback: true }
         );
-
-        if (isPrinted) {
-            this.currentOrder._printed = true;
-        }
```
[the remark this answers is not in the document]
Comment (vlst-odoo) [reply addons/point_of_sale/static/src/app/screens/receipt_screen/receipt_screen.js:135]: i restored the old version

File: addons/pos_restaurant/static/src/app/bill_screen/bill_screen.js (hunk 8 above)
Comment (vlst-odoo) [reply addons/pos_restaurant/static/src/app/bill_screen/bill_screen.js:25]: this code was the same in the bill screen as in the receipt screen so i lifted the function to the pos store

Comment (vlst-odoo) [reply addons/pos_restaurant/static/src/app/bill_screen/bill_screen.js]: that's true. Now it should have the same behavior as before

File: addons/pos_restaurant/static/src/overrides/models/pos_store.js [hunk 13]
```diff
@@ -49,8 +49,8 @@ patch(PosStore.prototype, {
         }
     },
     async actionAfterIdle() {
+        // FIXME: the open modals should be closed when the idle timer is triggered
         if (document.querySelector(".modal-open")) {
```
Comment (caburj) [inline addons/pos_restaurant/static/src/overrides/models/pos_store.js:53]: This condition here is a problem. It should be "if not error modal open, we should proceed with the logic". However, this is preventing the logic to run because most of the time, there is no modal open.
Please fix this in this PR (in a different commit), referencing the bug was introduced in the dialog PR.

File: addons/web/static/src/core/dialog/dialog.js [hunk 14]
```diff
@@ -98,7 +98,7 @@ export class Dialog extends Component {
             ref: this.modalRef,
             elements: ".modal-content",
             handle: ".modal-header",
-            ignore: "button",
+            ignore: "button, input",
```
Comment (vlst-odoo) [inline addons/web/static/src/core/dialog/dialog.js:101]: Hello, @odoo/rd-framework-js , do you agree with this change?

Comment (aab-odoo) [reply addons/web/static/src/core/dialog/dialog.js:101]: Looks reasonable (not sure about having inputs in the header though, but if that's what we want...)

File: addons/point_of_sale/static/src/app/screens/partner_list/partner_list.xml (hunk 10 above)
[the remark this answers is not in the document]
Comment (vlst-odoo) [reply addons/point_of_sale/static/src/app/screens/partner_list/partner_list.xml]: this change caused a stupid problem...
`onClickUnselect` passes the pointer event to it's handler, so the partner would become this object instead of just undefined, which was causing other problems. I reverted the change

**Comment by caburj** [comment]
robodoo r+

**Comment by caburj** [comment]
robodoo rebase-ff
